force-app/main/default/classes/PS_IntegrationServiceConfig.cls

```diff
@@ -74,26 +77,48 @@ public with sharing class PS_IntegrationServiceConfig {
 
     /**
     * @description Determines if the Elevate integration service is configured and enabled.
-    * The Elevate Payment Services integration is considered "enabled" (for now) when there
-    * is at least a single record in the protected Payment Services Configuration object
+    * The Elevate Payment Services integration is considered enabled when all required
+    * service records are present with value and isActive record is either missing or true
     * @return Boolean
     */
-    private Boolean isEnabled {
-        get {
-            if (isEnabled == null) {
-                return config.keyValueMap.keySet().containsAll(REQUIRED_CONFIG_KEYS);
-            }
-            return isEnabled;
-        }
-        set;
+    public Boolean isIntegrationEnabled() {
+        return isIntegrationActive() && hasElevateServiceData();
     }
 
     /**
-    * @description Returns the Elevate integration service enablement status
+    * @description Determines if the current user has the appropriate permissions to modify Elevate records
     * @return Boolean
     */
-    public Boolean isIntegrationEnabled() {
-        return isEnabled;
+    public Boolean hasIntegrationPermissions() {
+        return hasPermissions;
+    }
+
+    /**
+    * @description Determine if isActive service record exist. If the record exist,
+    * using the record to determine if Elevate is registered. Else assume the key
+    * is active
+    * @return Boolean
+    */
+    private Boolean isIntegrationActive() {
+        if (config.keyValueMap.containsKey(isActive)) {
```

Comment:
I'm glad you took this approach, we should only care about this key/value if a customer deactivated Elevate.

Reply:
What do y'all think about pushing this `config.keyValueMap.containsKey(isActive)` as a query method on config ala `config.hasIsActiveFlag()` and further can do `config.getIsActive()` instead of `Boolean.valueOf(config.keyValueMap.get(isActive))`?